engine/src/cmd/planet.cpp

```diff
@@ -573,7 +573,9 @@ void Planet::AddFog(const std::vector<AtmosphericFogMesh> &v, bool opticalillusi
     }
     fawg->setFaceCamera();
     getSubUnits().preinsert(fawg);
-    *fawg->current_hull /= fawg->GetHullPercent();
+
+    // TODO: this is very weird. Why are we doing this?
+    // fawg->hull /= fawg->GetHullPercent();
```

Comment:
I don't know why the code was doing that. If you don't know, leave it. It's one of those "fences." In my opinion, anyway.

Reply:
Let me clarify, as I do understand this code. 
fawg (why not fog?!) is a unit and created with some unspecified hull value, several lines up.
Now fawg = fawg * percent_damage.
So, example:
```
fawg = 50, percent = 50
new_fawg = 100
```
Except, this was created several lines above. Hull is undamaged. So x /= 1.0 is x

Reply:
Let's be generous and assume this unit creation is different and hull_percent != 1.
Why not just do:
`hull = x * some_percent`
Why be so convoluted?!

Reply:
So, digging a bit deeper (why though?!) I found this:
```
if (opticalillusion) {
   fawg = new AtmosphereHalo(this->rSize(), fogs, 0);
} else {
   fawg = new Unit(fogs, true, 0);
}


AtmosphereHalo(float radiusOfPlanet, vector<Mesh *> &meshes, int faction) :
    Unit(meshes, true, faction) {
    planetRadius = radiusOfPlanet;
}

Unit::Unit(std::vector<Mesh *> &meshes, bool SubU, int fact)
        : Drawable(), Damageable(), Movable()
```

So, it doesn't look like it even gets initialized with anything other than the default 1 value.

Reply:
So it looks like this is supposed to draw the atmosphere "halo" around a planet.

Reply:
It's a "Unit" because just about everything that needs to be drawn is either a Unit or a Bolt, I believe. Something like that. In this case, the atmosphere is drawn separately from the surface of the planet itself. As a separate, transparent layer. Same thing with the city lights, in cases where a planet has city lights. They constitute another layer.

Reply:
There are a lot of details that I still don't understand, but that seems to be the gist of it.

Reply:
Yep. You understood correctly

Also, we'd need @Loki1950 to comment on this but I'm reasonably certain none of these have really worked. Atmosphere, surface, nebula, etc.

Finally, this isn't how I'd implement this. I don't think firing a lot of drivers at the atmosphere is going to "boil" it.

Reply:
Um, what? I don't know about nebulae, but atmosphere and planet surface definitely work.

Reply:
Moving this to a discussion. We started from why `fawg->hull /= fawg->GetHullPercent();` and this is beyond the scope of this PR.

Reply:
@royfalk please link the discussion here

Reply:
https://github.com/orgs/vegastrike/discussions/977